fineract-e2e-tests-runner/src/test/resources/features/LoanCharge.feature

```diff
@@ -1673,3 +1672,39 @@ Feature: LoanCharge
+    Then Loan Repayment schedule has 1 periods, with the following data for periods:
+      | Nr | Days | Date              | Paid date          | Balance of loan | Principal due | Interest | Fees | Penalties | Due   | Paid  | In advance | Late | Outstanding |
+      |    |      | 27 September 2024 |                    | 40.0            |               |          | 0.0  |           | 0.0   | 0.0   |            |      |             |
+      | 1  | 30   | 27 October 2024   |                    | 0.0             | 40.0          | 0.33     | 0.0  | 0.0       | 40.33 | 0.0   | 0.0        | 0.0  | 40.33       |
+    Then Loan Repayment schedule has the following data in Total row:
+      | Principal due | Interest | Fees | Penalties | Due   | Paid  | In advance | Late | Outstanding |
+      | 40.0          | 0.33     | 0.0  | 0.0       | 40.33 | 0.0   | 0.0        | 0.0  | 40.33       |
+    Then Loan Transactions tab has the following data:
+      | Transaction date  | Transaction Type | Amount | Principal | Interest | Fees | Penalties | Loan Balance |
+      | 27 September 2024 | Disbursement     | 40.0   | 0.0       | 0.0      | 0.0  | 0.0       | 40.0         |
+      | 27 September 2024 | Accrual          | 0.33   | 0.0       | 0.33     | 0.0  | 0.0       | 0.0          |
+    When Admin adds "LOAN_NSF_FEE" due date charge with "27 September 2024" due date and 1 EUR transaction amount
+    When Admin makes a charge adjustment for the last "LOAN_NSF_FEE" type charge which is due on "27 September 2024" with 1 EUR transaction amount and externalId ""
+    Then Loan has 40.32 outstanding amount
+    Then Loan Repayment schedule has 1 periods, with the following data for periods:
+      | Nr | Days | Date              | Paid date         | Balance of loan | Principal due | Interest | Fees | Penalties | Due   | Paid  | In advance | Late | Outstanding |
+      |    |      | 27 September 2024 |                   | 40.0            |               |          | 0.0  |           | 0.0   | 0.0   |            |      |             |
+      | 1  | 30   | 27 October 2024   |                   | 0.0             | 40.0          | 0.32     | 0.0  | 1.0       | 41.32 | 1.0   | 1.0        | 0.0  | 40.32       |
+    Then Loan Repayment schedule has the following data in Total row:
+      | Principal due | Interest | Fees | Penalties | Due   | Paid  | In advance | Late | Outstanding |
+      | 40.0          | 0.32     | 0.0  | 1.0       | 41.32 | 1.0   | 1.0        | 0.0  | 40.32       |
+    Then Loan Transactions tab has the following data:
+      | Transaction date  | Transaction Type  | Amount | Principal | Interest | Fees | Penalties | Loan Balance |
+      | 27 September 2024 | Disbursement      | 40.0   | 0.0       | 0.0      | 0.0  | 0.0       | 40.0         |
+      | 27 September 2024 | Accrual           | 0.33   | 0.0       | 0.33     | 0.0  | 0.0       | 0.0          |
```

Comment:
Why do we have 2 accruals? Its weird...

Reply:
Ohh.. i see. Accounting is set to None... so it is creating a new one immediately for each event.